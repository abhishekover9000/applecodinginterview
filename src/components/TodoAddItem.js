import React, { Component } from "react";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";

class TodoAddItem extends Component {
  state = {
    name: ""
  };

  handleChange = e => {
    this.setState({ name: e.target.value });
  };

  add = () => {
    this.props.addItem(this.state.name);
  };
  render() {
    return (
      <div>
        <TextField
          value={this.state.name}
          label="Title"
          onChange={this.handleChange}
        />
        <Button onClick={this.add} variant="extendedFab" color="primary">
          {" "}
          Add{" "}
        </Button>
      </div>
    );
  }
}
export default TodoAddItem;
