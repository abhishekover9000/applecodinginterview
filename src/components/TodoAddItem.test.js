import React from "react";
import { shallow, configure } from "enzyme";
import TodoAddItem from "./TodoAddItem";

describe("TodoAddItem", () => {
  const app = shallow(<TodoAddItem />);

  it("renders properly", () => {
    expect(app).toMatchSnapshot();
  });
});
