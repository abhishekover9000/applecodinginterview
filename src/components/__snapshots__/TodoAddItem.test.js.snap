// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoAddItem renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoAddItem />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TextField
          label="Title"
          onChange={[Function]}
          required={false}
          select={false}
          value=""
          variant="standard"
        />,
        <WithStyles(Button)
          color="primary"
          onClick={[Function]}
          variant="extendedFab"
        >
           
          Add
           
        </WithStyles(Button)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "label": "Title",
          "onChange": [Function],
          "required": false,
          "select": false,
          "value": "",
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            " ",
            "Add",
            " ",
          ],
          "color": "primary",
          "onClick": [Function],
          "variant": "extendedFab",
        },
        "ref": null,
        "rendered": Array [
          " ",
          "Add",
          " ",
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TextField
            label="Title"
            onChange={[Function]}
            required={false}
            select={false}
            value=""
            variant="standard"
          />,
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
            variant="extendedFab"
          >
             
            Add
             
          </WithStyles(Button)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "label": "Title",
            "onChange": [Function],
            "required": false,
            "select": false,
            "value": "",
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              "Add",
              " ",
            ],
            "color": "primary",
            "onClick": [Function],
            "variant": "extendedFab",
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Add",
            " ",
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
